packages/block-editor/src/hooks/color.scss

```diff
@@ -0,0 +1,9 @@
+.color-block-support-panel {
+	.block-editor-contrast-checker {
+		grid-column: span 2;
```

Comment:
Is this change necessary after the latest changes to `ToolsPanel`'s layout?

Reply:
The contrast checker [isn't added as a `ToolsPanelItem`](https://github.com/WordPress/gutenberg/blob/update/color-support-to-use-tools-panel/packages/block-editor/src/hooks/color-panel.js#L89-L94) and as such doesn't get the [full-width style](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/tools-panel/styles.ts#L22-L24) by default.  This could possibly be removed when we settle on what `Grid` or `GridItem` props to expose for the `ToolsPanel`.

Reply:
> The contrast checker isn't added as a ToolsPanelItem

I'm probably missing something, but why can't it be added as a `ToolsPanelItem`?

Reply:
> I'm probably missing something, but why can't it be added as a ToolsPanelItem?

Adding it as a `ToolsPanelItem` means it would be displayed within the `ToolsPanel` menu. It isn't something that the user should be toggling on or off. 

If the contrast checker is wrapped up inside a `ToolsPanelItem` with another control say background color, it wouldn't appear in the menu but it wouldn't show when it should. Take the scenario where the user only toggles on the font color control and not the background color. They change that font color to something that should trigger the warning from the contrast checker however because the contrast checker is wrapped with the background color and that isn't displayed, the contrast checker remains hidden.

Duplicating the contrast checker so one each is wrapped in the color controls would mean when both font and background color are toggled on for display the user would get duplicate warnings. 

I believe this simple CSS adjustment for something that is related specifically to this panel should be ok.

Reply:
Thank you for the explanation, now I understand the situation much better.

> I believe this simple CSS adjustment for something that is related specifically to this panel should be ok.

That definitely sounds sensible for this PR!

I just didn't want to miss an interesting opportunity to test potential limitations of the current `ToolsPanel` component — do you think we could improve `ToolsPanel` to handle this situation better? (again, to be tackled in a separate PR)

Reply:
> do you think we could improve ToolsPanel to handle this situation better?

No great ideas come to mind. 

In terms of styling, we either need to:
- add some generic default that might not apply in all cases to the `ToolsPanel`, 
- stuff the `ToolsPanel` full of styles for random components,
- add a style for ad hoc components ( as done here ), 
- or, wrap it in a component that provides that style or equivalent props e.g. `grid-column`.

When switching the `ToolsPanel` to use the `Grid` component to handle its layout, it was decided we shouldn't expose extra props or add a `GridItem` like component. I would expect a case like this to be covered when that decision gets revisited.

Reply:
> When switching the ToolsPanel to use the Grid component to handle its layout, it was decided we shouldn't expose extra props or add a GridItem like component. I would expect a case like this to be covered when that decision gets revisited.

Yeah, that is definitely clear. I think that adding a style for ad hoc components ( as done here ) is the best choice at the moment.

---

The aspect that I was highlighting here is that we are displaying a component in the `ToolsPanel`  (the contrast checker) that can't be wrapped in a `ToolsPanelItem`, since it needs to be always visible (and should not appear in the `ToolsPanel` menu) — I was wondering if this feels hacky, but after reviewing how `ToolsPanel` is supposed to work, it looks like a perfectly fine pattern. At most, we could add a similar case scenario in the `ToolsPanel` readme to explain that items that are not wrapped in `ToolsPanelItem` are still going to be rendered as expected.

Definitely not a blocker for this PR in any case!

Reply:
>  I was wondering if this feels hacky, but after reviewing how ToolsPanel is supposed to work, it looks like a perfectly fine pattern

This aspect of the `ToolsPanel` design was intended and was suggested as feedback in some of the earliest iterations of the component. 

From memory, the earliest incarnation converted all immediate panel children to items in the menu. The shift to allowing panel items to register themselves opened up the possibility that not everything had to be included in the menu. 

My thinking at the time, was that we might have intro or summary text in a panel. Something similar to some of the Global Styles panels. I wasn't anticipating the display of those elements in the panel to be related to the items that were controlled via the menu but it's not a big leap.

<img width="278" alt="Screen Shot 2021-11-17 at 4 56 30 pm" src="https://user-images.githubusercontent.com/60436221/142149548-4c1b8054-2303-4f61-b334-df4a1a9c6711.png">

> At most, we could add a similar case scenario in the ToolsPanel readme to explain that items that are not wrapped in ToolsPanelItem are still going to be rendered as expected.

I'm happy to update the `ToolsPanel` readme and stories to include something that demonstrates that items not wrapped in a `ToolsPanelItem` are going to be rendered and is expected.

Reply:
PR improving the `ToolsPanel` readme and stories is available in: https://github.com/WordPress/gutenberg/pull/36557

Reply:
Thank you for the explanation and additional context, it sounds good to me — we can keep things as they currently are. I just wanted to make sure that the way this worked was "expected" and I lacked the context that you shared!

We can mark this conversation as solved :)